src/main/java/com/verily/genomewarp/utils/GenomeRangeUtils.java

```diff
@@ -0,0 +1,492 @@
+package com.verily.genomewarp.utils;
+
+import com.google.genomics.v1.Range;
+import com.verily.genomewarp.GenomeWarpSerial;
+import com.verily.genomewarp.HomologousRangeOuterClass.HomologousRange;
+import com.verily.genomewarp.HomologousRangeOuterClass.HomologousRange.TargetStrand;
+import com.verily.genomewarp.utils.GenomeWarpUtils.HomologousRangeComparator;
+import htsjdk.variant.variantcontext.VariantContext;
+import htsjdk.variant.vcf.VCFFileReader;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.SortedMap;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.regex.Matcher;
+
+public class GenomeRangeUtils {
+
+  private static final Logger logger = Logger.getLogger(GenomeRangeUtils.class.getName());
+
+  /**
+   * Splits a given set of genome ranges at non-DNA characters
+   *
+   * <p> This function reads in ranges using a BufferedReader and splits
+   * the given range along non-strict DNA characters (non ACTGactg).
+   *
+   * @param refFasta the reference genome, containing the string bases
+   * @param bed a BufferedReader containing the range reads
+   * @return a hashmap of chromosome to arraylist of GenomeRanges only containing valid DNA characters
```

Comment:
s/hashmap/TreeMap
s/arraylist/ArrayList

Reply:
There are ample of typos, inconsistencies in the docs, comments in the code and error messages. Not all of them are made by me and I'm not going to fix all of them (or more like while I will be fixing them, I will probably add more one way or another and there are some places in the code that I haven't even touched and I'm sure they also have some typos).
Is it really that important to fix every single one of them? It's like fighting windmills, I am trying to fix them when I see them, I did a few doc cleanups and typo fixes commits but I am not nitpicky enough to pursue and kill every single typo.

Reply:
It's more important that you feel happy while working, as then you'll produce your best work.  Everyone goes through this avalanche of inherited work, and it's just a balance of managing a progressive path.  If there are portions of the code that a majority of users will interface with, you will get proportionally less support-related questions with cleaner code.  If there are API developers who want to integrate with your work, they might be more picky and might prefer clean, well-documented APIs.  It's really a support-vs-development trade-off game of that each user-facing application goes through.  As your code will get more and more use, you can tackle any pain points incrementally as they come down the pipeline.  If you get any spare cycles down the line, you can always code-review and submit PRs that polishes the codebase, which becomes more of an advertisement to the world of presenting your work.